modules/nlu/src/backend/pipelines/entities/pattern_extractor.test.ts

```diff
@@ -127,7 +127,88 @@ I'm riding my mercedes-benz to the dealership then I will take my BM to buy an o
       expect(entities[3].meta.end).toEqual(139)
       expect(entities[3].meta.source).toEqual('BMW!')
       expect(entities[3].data.value).toEqual('BMW')
-    })
+    }),
+      test('Extract fuzzier list entities', async () => {
+        const entityDef = {
+          id: '_',
+          name: 'Oeufs',
+          type: 'list',
+          fuzzy: true,
+          occurences: [
+            {
+              name: 'mirroirs',
+              synonyms: []
+            },
+            {
+              name: 'brouillés',
+              synonyms: []
+            },
+            {
+              name: "dans l'trou",
+              synonyms: []
+            },
+            {
+              name: 'omelette',
+              synonyms: []
+            },
+            {
+              name: 'au fromage',
+              synonyms: []
+            }
+          ]
+        } as sdk.NLU.EntityDefinition
+
+        const userInput =
+          "J'aime les oeufs bourille, les oeufs au miroirr, les oeufs dan ltrou, les omelets et les oeufs au fomage"
+        //                  xxxxxxxxx              xxxxxxxx           xxxxxxxxxx     xxxxxxx              xxxxxxxxx
+        // 01234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123
+        // 00        10        20        30        40        50        60        70        80        90        100
+
+        const extractor = new PatternExtractor(Toolkit, languageProvider)
+
+        const sanitized = userInput.replace('\n', '')
+        const ds = initNLUStruct(sanitized, [], ['global'])
+        ds.sanitizedText = sanitized
+        ds.sanitizedLowerText = sanitized.toLowerCase()
+        const [stringTokens] = await languageProvider.tokenize([sanitized], 'en')
```

Comment:
for testing purposes, you could have mocked the language provider, it would skip http calls thus making tests run much faster

Reply:
nvm I just realized that you actually use a fake